design-documents/graph-ql/coverage/shared-cart.md

```diff
@@ -1,30 +1,45 @@
-| 3    | Smartphone | customerCartId()                                                                 | customer-token | 123e4567              |
-| 4    | Smartphone | getCart("123e4567") {cart_items {sku, quantity}}                                 | customer-token | {cart_items: [{"sku": "productA", "quantity": 2]} |
+| 1    | Laptop     | customerCartId()                                                                 | customer-token | CustomerCart123e4567              |
+| 2    | Laptop     | addProductsToCart("CustomerCart123e4567", [{"sku": "productA", "quantity": 2}]) {cart_id}    | customer-token | CustomerCart123e4567              |
+| 3    | Smartphone | customerCartId()                                                                 | customer-token | CustomerCart123e4567              |
+| 4    | Smartphone | getCart("CustomerCart123e4567") {cart_items {sku, quantity}}                                 | customer-token | {cart_items: [{"sku": "productA", "quantity": 2]} |
+
 
+ 2. **Merge guest cart:** There should be a way to transfer/merge products from a guest cart to a the customer cart. To achieve this we choose to add a mutation that accepts an active guest cart Id, and reflected merged products will be shown directly in the cart response object.
+ Proposal for this scenario reflected in schema is the following query:
+```graphql
+mergeGuestIntoCustomerCart(guestCartId: String): Cart! // where Cart! is the customer cart
+```
+Note: This operation will only work for registered customers and existing guest carts (valid customer token must be provided in headers). Also, this way se segregate operations as `customerCartId()` as query and `importGuestCart()` as mutation and their intent is very clear. A developer can choose 1st or the 2nd operation depending if the user had products in the cart or not. In both situations you will get the Customer Cart . Also see alternative for #2.
+  
 
  ## Alternatives
  
- - It is possible to make `cartId` optional argument and rely on `getCart {cart_id}` query for fetching customer cart ID. When used by customer, ID is not required. When used by guest - ID is required.
+ 1. It is possible to make `cartId` optional argument and rely on `getCart {cart_id}` query for fetching customer cart ID. When used by customer, ID is not required. When used by guest - ID is required.
     - guest still need to use `createEmptyCart` to obtain cart ID
     - `optional` argument makes semantics less clear
+    
+ 2. It is possible to achieve the merging of guest cart by adding an optional argument to `customerCart`, called `guestCartToMerge`. When used it will merge the guest cart into the Customer Cart. This makes the API not very clear, joins a query and a mutation together and could confuse the developer to understand it's intent. The benefit it adds is that we only need 1 call for both operations as converting guest cart into customer cart is a common operation but executed just one time per login.
```

Comment:
can you add an example of how this would look, either in the schema or a sample query